streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -100,37 +101,85 @@
     KStream<K, V> through(String topic);
 
     /**
+     * Materialize this stream to a topic, also creates a new instance of {@link KStream} from the topic
```

Comment:
For functions that include the StreamPartitioner instance, better add "... and a customizable {@link StreamPartitioner} to determine the distribution of records to partitions".